azurerm/helpers/azure/batch_pool.go

```diff
@@ -203,10 +204,44 @@ func FlattenBatchPoolContainerConfiguration(armContainerConfiguration *batch.Con
 	if armContainerConfiguration.Type != nil {
 		result["type"] = *armContainerConfiguration.Type
 	}
+	result["container_registries"] = flattenBatchPoolContainerRegistries(armContainerConfiguration.ContainerRegistries)
 
 	return []interface{}{result}
 }
 
+func flattenBatchPoolContainerRegistries(armContainerRegistries *[]batch.ContainerRegistry) []interface{} {
+	results := make([]interface{}, 0)
+
+	if armContainerRegistries == nil {
+		return results
+	}
+	for _, armContainerRegistry := range *armContainerRegistries {
+		result := flattenBatchPoolContainerRegistry(&armContainerRegistry)
+		results = append(results, result)
+	}
+	return results
+}
+
+func flattenBatchPoolContainerRegistry(armContainerRegistry *batch.ContainerRegistry) map[string]interface{} {
+	result := make(map[string]interface{})
+
+	if armContainerRegistry == nil {
+		return result
+	}
+	if registryServer := armContainerRegistry.RegistryServer; registryServer != nil {
+		result["registry_server"] = *registryServer
+	}
+	if userName := armContainerRegistry.UserName; userName != nil {
+		result["user_name"] = *userName
+	}
+	// We won't usually get the password back, but we include it here anyways
```

Comment:
If this is usually not returned we should not be setting it. What you can do is re use the password from state that you could get via `d.Get()`